services/checks/funcs.go

```diff
@@ -80,3 +85,79 @@ func formatVersionNum(args ...interface{}) (interface{}, error) {
 	}
 	return p.String(), nil
 }
+
+// GetAdditionalContext returns additional functions to be used in check scripts.
+func GetAdditionalContext() map[string]starlark.GoFunc {
+	return map[string]starlark.GoFunc{
+		"ip_is_private":      ipIsPrivate,
+		"parse_version":      parseVersion,
+		"format_version_num": formatVersionNum,
+	}
+}
+
+// ipIsPrivate accepts a single string argument (IP address) and
+// returns true for a private address, otherwise false.
+func ipIsPrivate(args ...interface{}) (interface{}, error) {
+	if l := len(args); l != 1 {
+		return nil, errors.Errorf("expected 1 argument, got %d", l)
+	}
+
+	ip, ok := args[0].(string)
+	if !ok {
+		return nil, errors.Errorf("expected string argument, got %[1]T (%[1]v)", args[0])
+	}
+
+	ipAddress := net.ParseIP(ip)
+	if ipAddress == nil {
+		// check if string was in CIDR notation
+		_, net, err := net.ParseCIDR(ip)
+		if err != nil {
+			return nil, errors.Errorf("invalid ip address: %s", ip)
+		}
+		for _, network := range privateNetworks {
+			if reflect.DeepEqual(net, network) {
+				return true, nil
+			}
+		}
+		return false, nil
+	}
+
+	for _, network := range privateNetworks {
+		if network.Contains(ipAddress) {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func init() {
+	// full list of reserved network addresses https://en.wikipedia.org/wiki/Reserved_IP_addresses
+	privateAddressBlocks := []string{
+		// private blocks, see https://tools.ietf.org/html/rfc1918
+		"10.0.0.0/8",
+		"172.16.0.0/12",
+		"192.168.0.0/16",
+		// shared address space range, see https://tools.ietf.org/html/rfc6598#section-7
+		"100.64.0.0/10",
+		// IETF protocol assignments range, see https://tools.ietf.org/html/rfc5735#section-4
+		"192.0.0.0/24",
+		// see https://tools.ietf.org/html/rfc5735#section-4
+		"198.18.0.0/15",
```

Comment:
added 3 more ipv4 address ranges that are classified as private networks https://en.wikipedia.org/wiki/Reserved_IP_addresses

had missed these the first time

Reply:
We probably can link straight to Wikipedia instead all those RFCs :)

Will it be easier to use net.IP method like IsLoopback there?

Reply:
> We probably can link straight to Wikipedia instead all those RFCs :)

done 

> Will it be easier to use net.IP method like IsLoopback there?

I would rather check for all the address ranges the same way rather than using a special method just for the loopback blocks. If that is what you mean by this comment.

Reply:
There are several methods: IsLoopback, IsLinkLocalMulticast, IsLinkLocalUnicast, etc.
I'm okay with not using them, but please check they all are covered correctly

Reply:
added multicast interface-local and multicast link-local